### PR TITLE
Basic functional (function based) multitasking

### PR DESCRIPTION
**Pull request by Kfeavel**

### CODE REVIEW

File: src/drivers/DriverManager.cpp [hunk 1]
```diff
@@ -24,3 +24,21 @@ void DriverManager::activateAll() {
...
+void DriverManager::activateLast() {
+    int index = numberOfDrivers - 1;
+    drivers[index]->activate();
+}
+
+Driver* DriverManager::getDriverWithTag(char* tag) {
+    // Cycle through the drivers which have been added
+    for (int i = 0; i < numberOfDrivers; i++) {
+        // If we find a matching tag
+        if (drivers[i]->getDriverTypeTag() == tag) {
```
Comment (micahswitzer) [inline src/drivers/DriverManager.cpp]: Does this work? This looks like it's comparing the pointers and not the actual contents of the string.

**Review by micahswitzer — APPROVED** [review]
Looks pretty good! Make sure you check out my one comment.

File: src/drivers/DriverManager.cpp (hunk 1 above)
Comment (micahswitzer) [reply src/drivers/DriverManager.cpp]: Maybe it would be better to use an enum?

Comment (Kfeavel) [reply src/drivers/DriverManager.cpp]: I thought about using an enum. We should open an issue for an optimization later.

**Pull request merged**